template/ts-template/test/Lock.ts

```diff
@@ -0,0 +1,220 @@
+  const deployContract = async (
+    contractName: string,
+    constructorArgs: ContractConstructorArgs<typeof LockContract.abi>,
+    txOptions: PayableCallOptions = {}
+  ) => {
+
+    let txOptionsObj = txOptions;
+
+    // use first default account if not provided explicitly for sending transaction
+    if (web3.utils.isNullish(txOptionsObj.from))
+      txOptionsObj = {
+        from: (await web3.eth.getAccounts())[0],
+        ...txOptionsObj
+      };
+
+    const artifact: Artifact = await hre.artifacts.readArtifact(contractName);
+
+    const contract = new web3.eth.Contract(artifact.abi);  // this will not have full types support 
+
+    const deployedContract = await contract.deploy({
+      data: artifact.bytecode,
+      arguments: constructorArgs
+    }).send(txOptionsObj);
+
+    deployedContract.defaultAccount = txOptionsObj.from; // this will be used for sending future transactions to contract for all .send() function calls 
+    return deployedContract;
+  }
+
+  //TODO: add  @chainsafe/hardhat-ts-artifact-plugin output support in Artifacts object in HRE and use above function for reading abi from there with full types support
+  const deployLockContract = async (
+    constructorArgs: ContractConstructorArgs<typeof LockContract.abi>,
+    txOptions: PayableCallOptions = {}
+  ) => {
+
+    let txOptionsObj = txOptions;
+    const contractName = "Lock";
+
+    // use first default account if not provided explicitly for sending transaction
+    if (web3.utils.isNullish(txOptionsObj.from))
+      txOptionsObj = {
+        from: (await web3.eth.getAccounts())[0],
+        ...txOptionsObj
+      };
+
+    const artifact: Artifact = await hre.artifacts.readArtifact(contractName);
+
+    // For exact types support we will use @chainsafe/hardhat-ts-artifact-plugin by npx hardhat ts-artifact after npx hardhat compile and pass that to abi function param
+    const contract = new web3.eth.Contract(LockContract.abi);
+
+    const deployedContract = await contract.deploy({
+      data: artifact.bytecode,
+      arguments: constructorArgs
+    }).send(txOptionsObj);
+
+    deployedContract.defaultAccount = txOptionsObj.from; // this will be used for sending future transactions to contract for all .send() function calls 
+    return deployedContract;
+  }
+
+  async function deployOneYearLockFixture() {
+    const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
+    const ONE_GWEI = 1_000_000_000;
+
+    const lockedAmount = ONE_GWEI;
+    const unlockTime = BigInt((await time.latest()) + ONE_YEAR_IN_SECS);
+
+    // Contracts are deployed using the first signer/account by default
+    const [owner, otherAccount] = await web3.eth.getAccounts();
+
+    const lock = await deployLockContract([unlockTime], {
+      value: String(lockedAmount),
+    });
+
+    return {
+      lock,
+      unlockTime,
+      lockedAmount,
+      owner,
+      otherAccount,
+      web3
+    };
+  }
+
+  describe("Deployment", function () {
```

Comment:
Is there a reason why the tests are not in separate files?

Reply:
Currently these are grouped in a file by contract name and using fixtures from same file, just for simplicity for starting.